plots/plot.py

```diff
@@ -103,178 +106,50 @@ def printSummaryStats(dataset):
-        i += 1
-        j -= 1
-    plot.write(lines)
-    plot.close()
-    os.system("gnuplot -c wealth.plg {0}".format(outfile))
-
-def generateTotalWealthNormalizedPlot(models, outfile):
-    print("Generating total wealth normalized plot script")
-    plot = open("wealth_normalized.plg", 'w')
-    config = "set xlabel \"Timestep\"\nset ylabel \"Total Wealth / Population\"\nset lt 1 lw 2 lc \"black\"\n"
-    config += "set xtics nomirror\nset ytics nomirror\nset key fixed right bottom\nset term pdf font \"Times,20\"\nset output \"wealth_normalized.pdf\"\n\n"
-    plot.write(config)
-    i = 0
-    j = len(models) - 1
-    lines = ""
-    for model in dataset:
-        if i == 0:
-            lines += "plot ARGV[1] using 'timestep':(column('{0}_welt')/column('{0}_pop')) with linespoints pointinterval 100 pointsize 0.75  lc 'magenta' lt 1 dt 1 pt {1} title 'Utilitarian', \\\n".format(model, j)
-        elif i < len(models) - 1:
-            lines += "\t'' u 'timestep':(column('{0}_welt')/column('{0}_pop')) with linespoints pointinterval 100 pointsize 0.75  lc 'cyan' lt 1 dt 1 pt {1} title 'Egoist', \\\n".format(model, j)
-        else:
-            lines += "\t'' u 'timestep':(column('{0}_welt')/column('{0}_pop')) with linespoints pointinterval 100 pointsize 0.75  lc 'gold' lt 1 dt 1 pt {1} title 'Raw Sugarscape'".format(model, j)
-        i += 1
-        j -= 1
-    plot.write(lines)
-    plot.close()
-    os.system("gnuplot -c wealth_normalized.plg {0}".format(outfile))
-
-def generateStarvationAndCombatPlot(models, outfile):
-    print("Generating starvation and combat deaths plot script")
-    plot = open("deaths.plg", 'w')
-    config = "set xlabel \"Timestep\"\nset ylabel \"Deaths / Population\"\nset lt 1 lw 2 lc \"black\"\n"
-    config += "set xtics nomirror\nset ytics nomirror\nset key fixed right top\nset term pdf font \"Times,24\"\nset output \"deaths.pdf\"\n\n"
-    plot.write(config)
-    i = 0
-    j = len(models) - 1
-    lines = ""
+        print("Generating total wealth plot")
+        generateSimpleLinePlot(models, dataset, totalTimesteps, "wealth.pdf", "agentWealthTotal", "Total Wealth", "center right")
+
+def generateSimpleLinePlot(models, dataset, totalTimesteps, outfile, column, label, positioning, percentage=False):
+    matplotlib.pyplot.rcParams["font.family"] = "serif"
+    matplotlib.pyplot.rcParams["font.serif"] = ["Times New Roman"]
+    matplotlib.pyplot.rcParams["font.size"] = 20
+    figure, axes = matplotlib.pyplot.subplots()
+    axes.set(xlabel = "Timestep", ylabel = label, xlim = [0, totalTimesteps])
+    x = [i for i in range(totalTimesteps + 1)]
+    lines = []
+    colors = ["magenta", "cyan", "gold", "black"]
+    c = 0
+    modelStrings = {"bentham": "Utilitarian", "egoist": "Egoist", "altruist": "Altruist", "none": "Raw Sugarscape", "rawSugarscape": "Raw Sugarscape"}
```

Comment:
We are allowing multiple decision models per simulation in our data collection config but not handling the naming correctly. This line along with line 142 throws `KeyError` when multiple decision models are in the same simulation.

Reply:
For line 142, you can check if the model is a list, and use `' '.join(modelStrings[model] for model in modelList)` if it is. This will make the names pretty long. My solution to this in my old multiple decision models branch was to abbreviate, e.g. `benthamHalfLookaheadBinary` -> `bHLB`.

Reply:
This PR is a drop-in replacement for the old plotting script, which also did not accurately plot mixed decision models or the long name ones. This PR now fixes that bug while also sidestepping this issue for now. The plotting script has always been a more bespoke tool for our publications. That's the primary goal for #46: make this a general purpose tool for users to query their dataset and produce plots. This is a significant step toward that goal.